printers/printerTree.php

```diff
@@ -135,36 +137,50 @@ private function str_contains(string $haystack, string $needle){
         return strpos($haystack, $needle) !== false;
     }
 
-    private function _fillTree($tree, $keys, $item, $parentId) {
+    private function _fillTree($tree, $keys, $item, $parentId, $myNs) {
         if (empty($keys)){ // We've reach the end of the journey. We register the data of $item
             if($item['type'] === 'd') {
                 $tree->self = $item;
             } else {
-                $tree->pages []= $item;
+                if ($myNs == $item['id']) {
```

Comment:
This line is needed, because I treat all nodes that same (you distinguished "pages" and "childrens"), "f" node can also had childrens because "f" node may it be e.g. "ns:start" page. In my wiki I have NS "tool:proxmox" that it has some pages (eg: installation, pbs), but I also have page "tool:proxmox" (page with tricks for pve) . Without this trick I have duplicated entries in tree, with this line I have only one correct entry.

I tested it on my local instance with AlmaLinux9 and php 8.2, everything worked (1:1 copy of my production wiki). When I move this fix to my "production" dokuwiki (centos 7 + php 7.4) one of NS not woking. In tree was only link to main page of NS. There was no subpages. When I started debbuging I discovered that "f" node was given earlier than "d" node on PHP7.4 and C7. I added then few lines that should prevent from that situation in future.